lib/open_telemetry/tracer.ex

```diff
@@ -0,0 +1,74 @@
+defmodule OpenTelemetry.Tracer do
+  @moduledoc """
+  Defines a named tracer.
+
+      defmodule MyApp.Tracer do
+        use OpenTelemetry.Tracer,
+          name: :my_app
+      end
+
+  Then used in your application:
+
+      require MyApp.Tracer
+
+      MyApp.Tracer.with_span "span-1" do
+        ... do something ...
+      end
+
+  """
+
+  @type t :: module
+
+  @doc false
+  defmacro __using__(opts) do
+    quote bind_quoted: [opts: opts] do
+      # @behaviour OpenTelemetry.Tracer
+
+      @otp_app opts[:otp_app] || "" # can't we get this automatically?
```

Comment:
Nope!

Reply:
At least not really during compile time. Ugly hacks possible, but edge cases are too broad to make it feasible in any way.